### PR TITLE
Feature/uppsf 1912 add broader relation

### PR DESCRIPTION
**Pull request by ivan-p-nikolov**
# Description

## What

Add the functionally to provide broader relations to a concept.
Extend the client interface to get concepts by custom metadata field.

## Why

https://financialtimes.atlassian.net/browse/UPPSF-1912

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [X] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -61,8 +61,8 @@ func (c *Client) CreateConcept(ctx context.Context, concept Concept, task string
 		return errors.New("input concept should have prefLaber defined")
 	}
 
-	if concept.SchemaObject == "" {
-		return errors.New("input concept should have schema defined")
+	if concept.SchemaObject == "" && concept.Broader == "" {
```
Comment (epavlova) [inline client.go:64]: Question: is it true that when you give `broader` concept the current one will inherit the `schema` from the parent concept?

File: client.go [hunk 2]
```diff
@@ -130,6 +130,35 @@ func (c *Client) AddConceptMetadataField(ctx context.Context, conceptID, fieldNa
...
+func (c *Client) GetConceptsWithCustomMetadata(ctx context.Context, task string, field string, value string) ([]interface{}, error) {
+	params := url.Values{}
+	params.Add("path", path.Join(
+		fmt.Sprintf("task:%s:%s", c.model, task),
+		"skos:Concept",
+		"meta:transitiveInstance",
+	))
+	params.Add("properties", `rdf:type,meta:displayName,[]`)
+	params.Add("filters", fmt.Sprintf(`subject(<%s>="%s")`, field, value))
+	reqURL := c.baseAPIURL.String() + "?" + params.Encode()
```
Comment (epavlova) [inline client.go:142]: Only suggestion: take advantage of the `url` package:
```
reqURL := c.baseAPIURL
reqURL. RawQuery = params.Encode()
```

Then when you use the `String()` method on `reqURL` you will get everything concatenated.

**Review by epavlova — DISMISSED** [review]

File: client.go (hunk 1 above)
Comment (ivan-p-nikolov) [reply client.go:64]: The way I understand it, yes. This is from the documentation in the ontology editor: 
> In Ontology Editor, a concept is said to belong to a concept scheme if it is a child (directly or indirectly) of a top concept.

**Review by epavlova — APPROVED** [review]

File: concept.go [hunk 3]
```diff
@@ -51,10 +54,24 @@ func (c Concept) MarshalJSON() ([]byte, error) {
...
 			ID: c.SchemaObject,
-		},
+		}
 	}
+
+	if c.Broader != "" {
+		input.Broader = &conceptID{
+			ID: c.Broader,
+		}
+	}
```
Comment (karsov) [inline concept.go:73]: Is this the code that says SL to create the "broader" relation?

**Review by karsov — APPROVED** [review]
Looks OK, although it is hard to say without a usage example.

File: concept.go (hunk 3 above)
Comment (ivan-p-nikolov) [reply concept.go:73]: Yes. We provide concept uri to relate to.

**Pull request merged**